lib/vanagon/platform/windows.rb

```diff
@@ -53,9 +54,54 @@ def generate_packaging_artifacts(workdir, name, binding)
       # @param name [String] name of the project
       # @param binding [Binding] binding to use in evaluating the packaging templates
       def generate_msi_packaging_artifacts(workdir, name, binding)
-        FileUtils.mkdir_p(File.join(workdir, "wix"))
-        erb_file(File.join(VANAGON_ROOT, "resources/windows/wix/project.wxs.erb"), File.join(workdir, "wix",  "#{name}.wxs"), false, { :binding => binding })
-        erb_file(File.join(VANAGON_ROOT, "resources/windows/wix/project.filter.xslt.erb"), File.join(workdir, "wix", "#{name}.filter.xslt"), false, { :binding => binding })
+        # Copy the project specific files first
+        copy_from_project("#{@proj_root}/resources/windows/wix", workdir)
+        merge_defaults_from_vanagon("#{VANAGON_ROOT}/resources/windows/wix", "#{workdir}/wix")
+        process_templates("#{workdir}/wix", binding)
+      end
+
+      # Method to recursively copy from a source project resource directory
+      # to a destination (wix) work directory.
+      # strongly suspect the original cp_r command would have done all of this.
+      #
+      # @param proj_resources [String] Project Resource File directory
+      # @param destination [String] Destination directory
+      # @param verbose [String] True or false
+      def copy_from_project(proj_resources, destination, verbose: false)
+        FileUtils.cp_r(proj_resources, destination, :verbose => verbose)
+      end
+
+      # Method to merge in the files from the Vanagon (generic) directories.
+      # Project Specific files take precedence, so since these are copied prior
+      # to this function, then this merge operation will ignore existing files
+      #
+      # @param vanagon_root [String] Vanagon wix resources directory
+      # @param destination [String] Destination directory
+      # @param verbose [String] True or false
+      def merge_defaults_from_vanagon(vanagon_root, destination, verbose: false)
+        vanagon_path = Pathname.new(vanagon_root)
+        files = Dir.glob(File.join(vanagon_root, "**/*.*"))
+        files.each do |file|
```

Comment:
Oh, thinking this through further, what's the win to having this function do the `Dir.glob` instead of having it operate over a list of files passed to it and doing the glob outside of the function?